docs/src/configurations.md

```diff
@@ -113,8 +121,107 @@ julia> rc"[Ne] 3s2" ⊗ rcs"3p2"
 ## Spin configurations
 
 ```@docs
+SpinConfiguration
 spin_configurations
 substitutions
+@scs_str
+```
+
+## Excited configurations
+
+AtomicLevels.jl provides an easy interface for generating lists of
+configurations which are the result of exciting one or more orbitals
+of a reference set to a set of substitution orbitals. This is done
+with [`excited_configurations`](@ref), which provides various
+parameters for controlling which excitations are generated. A very
+simple example could be
+
+```jldoctest
+julia> excited_configurations(c"1s2", os"2[s-p]"...)
+4-element Array{Configuration{Orbital{Int64}},1}:
+ 1s²
+ 1s 2s
+ 2s²
+ 2p²
+```
+
+which as we see contains all configurations generated by at most
+exciting two orbitals `1s²` and keeping the overall parity. By lifting
+these restrictions, more configurations can be generated:
+
+```jldoctest
+julia> excited_configurations(c"1s2 2s", os"3[s-p]"...,
+                              keep_parity=false, max_excitations=2)
+14-element Array{Configuration{Orbital{Int64}},1}:
+ 1s² 2s
+ 1s 2s²
+ 1s 2s 3s
+ 1s 2s 3p
+ 1s² 3s
+ 1s² 3p
+ 2s² 3s
+ 2s² 3p
+ 2s 3s²
+ 2s 3s 3p
+ 1s 3s²
+ 1s 3s 3p
+ 2s 3p²
+ 1s 3p²
+
+julia> excited_configurations(c"1s2 2s", os"3[s-p]"...,
+                              keep_parity=false, max_excitations=3)
+17-element Array{Configuration{Orbital{Int64}},1}:
+ 1s² 2s
+ 1s 2s²
+ 1s 2s 3s
+ 1s 2s 3p
+ 1s² 3s
+ 1s² 3p
+ 2s² 3s
+ 2s² 3p
+ 2s 3s²
+ 2s 3s 3p
+ 1s 3s²
+ 1s 3s 3p
+ 2s 3p²
+ 1s 3p²
+ 3s² 3p
+ 3s 3p²
+ 3p³
+```
+
+Since configurations by default are unsorted, when exciting from
+[`SpinConfiguration`](@ref)s, the substitutions are performed
+in-place:
+
+```jldoctest
+julia> excited_configurations(first(scs"1s2"), sos"2[s-p]"...)
+21-element Array{Configuration{SpinOrbital{Orbital{Int64}}},1}:
+ 1s₀α 1s₀β
+ 2s₀α 1s₀β
+ 2s₀β 1s₀β
+ 1s₀α 2s₀α
+ 1s₀α 2s₀β
+ 2s₀α 2s₀β
+ 2p₋₁α 2p₋₁β
+ 2p₋₁α 2p₀α
+ 2p₋₁α 2p₀β
+ 2p₋₁α 2p₁α
+ 2p₋₁α 2p₁β
+ 2p₋₁β 2p₀α
+ 2p₋₁β 2p₀β
+ 2p₋₁β 2p₁α
+ 2p₋₁β 2p₁β
+ 2p₀α 2p₀β
+ 2p₀α 2p₁α
+ 2p₀α 2p₁β
+ 2p₀β 2p₁α
+ 2p₀β 2p₁β
+ 2p₁α 2p₁β
+```
```

Comment:
This doctest does not pass.. but I believe it's just different printing. Should be this I think:

```
21-element Array{Configuration{SpinOrbital{Orbital{Int64}}},1}:
 1s₀α 1s₀β
 2s₀α 1s₀β
 2s₀β 1s₀β
 1s₀α 2s₀α
 1s₀α 2s₀β
 2s₀α 2s₀β
 2p₋₁α 2p₋₁β
 2p₋₁α 2p₀α
 2p₋₁α 2p₀β
 2p₋₁α 2p₁α
 ⋮
 2p₋₁β 2p₀β
 2p₋₁β 2p₁α
 2p₋₁β 2p₁β
 2p₀α 2p₀β
 2p₀α 2p₁α
 2p₀α 2p₁β
 2p₀β 2p₁α
 2p₀β 2p₁β
 2p₁α 2p₁β
```

Reply:
I know, but the difference is only two elements, so let's keep it until Documenter supports tailoring the output length :)